contrib/libpcap/gencode.c

```diff
@@ -591,8 +591,8 @@ newchunk_nolongjmp(compiler_state_t *cstate, size_t n)
 	size_t size;
 
 #ifndef __NetBSD__
-	/* XXX Round up to nearest long. */
-	n = (n + sizeof(long) - 1) & ~(sizeof(long) - 1);
+	/* XXX Round up to nearest long or capability size. */
+	n = (n + MAX(sizeof(void *), sizeof(long)) - 1) & ~(MAX(sizeof(void *), sizeof(long)) - 1);
```

Comment:
In C11+ (which we can assume in CheriBSD) it should be `_Alignof(max_align_t)` not any of this, and for upstream it should be some C89-compatible `offsetof(struct { char dummy; union { long long x; void *p; } u; }, u)` thing (but not inline). I intend to just upstream a proper fix and then pull it back into FreeBSD+CheriBSD.

Reply:
https://github.com/the-tcpdump-group/libpcap/pull/1158